client/state/ui/drop-zone/reducer.js

```diff
@@ -17,10 +13,16 @@ import {
 
 import { createReducer } from 'state/utils';
 
-const isVisible = createReducer( false,
+const isVisible = createReducer( {},
 	{
-		[ DROPZONE_SHOW ]: stubTrue,
-		[ DROPZONE_HIDE ]: stubFalse,
+		[ DROPZONE_SHOW ]: ( state, { dropZoneName } ) => ( {
+			...state,
+			[ dropZoneName ]: true,
+		} ),
+		[ DROPZONE_HIDE ]: ( state, { dropZoneName } ) => ( {
+			...state,
+			[ dropZoneName ]: false,
+		} ),
```

Comment:
You could use `keyedReducer`. From the top of my mind:

```js
const isVisible = keyedReducer( 'dropZoneName', createReducer( false, {
  [ DROPZONE_SHOW ]: stubTrue,
  [ DROPZONE_HIDE ]: stubFalse,
} ) );
```

The only caveat would be that `keyedReducer`'d reducers don't handle `{DE,}SERIALIZE` actions (meaning this state would be persisted even though that doesn't make sense) *but* it's fine in this case because we're inside `state.ui`, whose base reducer has a [clause to avoid persistence](https://github.com/Automattic/wp-calypso/blob/c8e59d1687d40c54da394e5b7f1afe0f469b9510/client/state/ui/reducer.js#L99).